src/sequential/sequential.hpp

```diff
@@ -0,0 +1,198 @@
+#pragma once
+
+#include <cassert>
+#include <cstdint>
+#include <iostream>
+#include <utility>
+#include <vector>
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// TYPE DEFINITIONS (for easy refactor)
+////////////////////////////////////////////////////////////////////////////////
+
+/// N.B.  I use int32_t for the key and value so that they fit into a 64-bit word
+///       if I so choose to do that. This way, we can do atomic updates. The
+///       extra metadata is a bit of an oof though.
+/// N.B.  I do not use *_t because this is a reserved name in POSIX.
+/// N.B.  I also named them something unique so it's easy to find/replace.
+typedef int32_t KeyType;
+typedef int32_t ValueType;
+/// N.B.  I use 'HashCode' because I want to distinguish 'hash' (verb) and
+///       'hash' (noun). Thus, I use 'hash code' to denote the latter.
+typedef size_t HashCodeType;
```

Comment:
```suggestion
using KeyType = int32_t;
using ValueType = int32_t;
/// N.B.  I use 'HashCode' because I want to distinguish 'hash' (verb) and
///       'hash' (noun). Thus, I use 'hash code' to denote the latter.
using HashCodeType = size_t;
```
`using` reads more naturally, and avoids the confusion of stuff like
```c++
typedef Error (*HandlerList[])(Event event);
```
Compare with
```c++
using HanderList = Error (*[])(Event event);
```

Reply:
I like this. Very idiomatic